src/darwin/Framework/CHIP/templates/CHIPClustersObjc-src.zapt

```diff
@@ -16,131 +16,101 @@
-{{>CHIPCallbackBridge name="Int32sAttribute"      type="Int32s"      }}
-{{>CHIPCallbackBridge name="Int64uAttribute"      type="Int64u"      }}
-{{>CHIPCallbackBridge name="Int64sAttribute"      type="Int64s"      }}
+typedef CHIP_ERROR (^CHIPActionBlock)();
 
-{{#chip_client_clusters}}
-{{#chip_cluster_responses}}
-class CHIP{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}CallbackBridge : public Callback::Callback<{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}Callback>
-{
+class CHIPCallbackBridge {
 public:
-    CHIP{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}CallbackBridge(ResponseHandler handler, dispatch_queue_t queue): Callback::Callback<{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}Callback>(CallbackFn, this), mHandler(handler), mQueue(queue) {}
-
-    ~CHIP{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}CallbackBridge() {};
-
-    static void CallbackFn(void * context{{#chip_cluster_response_arguments}}, {{asUnderlyingZclType type}} {{asSymbol label}}{{/chip_cluster_response_arguments}})
-    {
-        CHIP{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}CallbackBridge * callback = reinterpret_cast<CHIP{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}CallbackBridge *>(context);
-        if (callback && callback->mQueue)
+    CHIPCallbackBridge(ResponseHandler handler, bool keepAlive = false)
+        : mHandler(handler)
+        , mKeepAlive(keepAlive)
         {
-            dispatch_async(callback->mQueue, ^{
-                callback->mHandler(nil, @{
-                  {{#chip_cluster_response_arguments}}
-                  {{#if isArray}}
-                  // {{asSymbol label}}: {{asUnderlyingZclType type}}
-                  // Conversion from this type to Objc is not properly implemented yet
-                  {{else if (isOctetString type)}}
-                  @"{{asSymbol label}}": [NSData dataWithBytes:{{asSymbol label}}.data() length:{{asSymbol label}}.size()],
-                  {{else if (isCharString type)}}
-                  @"{{asSymbol label}}": [NSString stringWithFormat:@"%s", {{asSymbol label}}],
-                  {{else}}
-                  @"{{asSymbol label}}": [NSNumber numberWith{{asObjectiveCNumberType label type false}}:{{asSymbol label}}],
-                  {{/if}}
-                  {{/chip_cluster_response_arguments}}
-                });
-                callback->Cancel();
-                delete callback;
-            });
+            chip::Callback::Callback<DefaultFailureCallback> * failureCallback = new chip::Callback::Callback<DefaultFailureCallback>(OnFailureFn, this);
+            Failure = failureCallback->Cancel();
         }
-    };
-
-private:
-    ResponseHandler mHandler;
-    dispatch_queue_t mQueue;
-};
 
-{{/chip_cluster_responses}}
-{{/chip_client_clusters}}
+    ~CHIPCallbackBridge()
+    {
+        Success->Cancel();
+        Failure->Cancel();
```

Comment:
Do you need the explicit `Cancel` calls?  I thought the whole point of using `Cancelable` is that its destructor cancels?